admin/class-primary-term-admin.php

```diff
@@ -105,7 +105,7 @@ protected function get_primary_term( $taxonomy_name ) {
 	protected function get_primary_term_taxonomies( $post_ID = null ) {
 
 		if ( null === $post_ID ) {
-			$post_ID = get_the_ID();
+			$post_ID = $_GET['post'];
```

Comment:
Wouldn't `get_queried_object_id()` be better here? 

And if not, please cast to `int` :)